diff -b -u -p -r -u src/gcc-3.4.6/gcc/collect2.c src/gcc-3.4.6/gcc/collect2.c
--- src/gcc-3.4.6/gcc/collect2.c	2005-01-10 16:25:23.000000000 +0100
+++ src/gcc-3.4.6/gcc/collect2.c	2011-01-11 14:33:07.000000000 +0100
@@ -1534,7 +1534,7 @@ collect_execute (const char *prog, char 
   if (redir)
     {
       /* Open response file.  */
-      redir_handle = open (redir, O_WRONLY | O_TRUNC | O_CREAT);
+      redir_handle = open (redir, O_WRONLY | O_TRUNC | O_CREAT, 0666);
 
       /* Duplicate the stdout and stderr file handles
 	 so they can be restored later.  */
